confgenerator/self_logs.go

```diff
@@ -113,14 +97,71 @@ func generateFluentBitSelfLogsComponents(ctx context.Context) []fluentbit.Compon
 			},
 		},
 	}.Components(ctx, fluentBitSelfLogsTag, "self-logs-severity")...)
-	out = append(out, fluentbit.TranslationComponents(fluentBitSelfLogsTag, "severity", severityKey, true,
-		[]struct{ SrcVal, DestVal string }{
-			{"debug", "DEBUG"},
-			{"error", "ERROR"},
-			{"info", "INFO"},
-			{"warn", "WARNING"},
-		})...,
-	)
+	out = append(out, LoggingProcessorModifyFields{
+		Fields: map[string]*ModifyField{
+			"severity": {
+				MoveFrom: "jsonPayload.severity",
+				MapValues: map[string]string{
+					"error": "ERROR",
+					"warn":  "WARNING",
+					"info":  "INFO",
+					"debug": "DEBUG",
+				},
+				MapValuesExclusive: true,
+			},
+		},
+	}.Components(ctx, fluentBitSelfLogsTag, "mapseverityvalues")...)
+	return out
+}
+
+type selfLogTranslationEntry struct {
+	fieldMatch string
```

Comment:
Are we going to match on any other field than the message?
Maybe we can simplify this by always matching on the message

Reply:
We are not going to match any other field really. I will remove it.